py/desimodel/install.py

```diff
@@ -32,36 +32,69 @@ def assert_svn_exists():
     except CalledProcessError as e:
         raise AssertionError("The svn command ({0}) on this system does not work. Output is: '{1}'.".format(e.cmd, e.output))
 
+def get_svn_version(desimodel_version=None):
+    """
+    Return which svn version should be checked out for given desimodel_version
+
+    Parameters
+    ----------
+    desimodel_version : :class:`str`, optional
+        The version X.Y.Z to download, trunk, or something of the
+        form branches/... Defaults to package version if x.y.z tagged,
+        otherwise trunk.
 
-def svn_export(desimodel_version=None):
+    Returns
+    -------
+    :class:`str`
+        svn version string
+    """
+
+    if desimodel_version is None:
+        from . import __version__ as this_version
+        # check for tag-like versions x.y.z or x.y
+        if (re.match(r'^\d+\.\d+\.\d+$', this_version) is not None or
+            re.match(r'\d+\.\d+$', this_version) is not None):
+            svn_version = 'tags/' + this_version
+        # otherwise use trunk if version wasn't specified
+        else:
+            svn_version = 'trunk'
+    elif (desimodel_version == 'trunk') or ('branches/' in desimodel_version):
+        svn_version = desimodel_version
+    else:
+        svn_version = 'tags/' + desimodel_version
+
+    return svn_version
+
+def svn_export(desimodel_version=None, svn_checkout=False):
     """Create a :command:`svn export` command suitable for downloading a
     particular desimodel version.
 
     Parameters
     ----------
     desimodel_version : :class:`str`, optional
         The version X.Y.Z to download, trunk, or something of the
-        form branches/... Defaults to trunk.
+        form branches/... Defaults to package version if x.y.z tagged,
+        otherwise trunk.
+    svn_checkout : bool, default False
+        If True, svn checkout instead of svn export
 
     Returns
     -------
     :class:`list`
         A :command:`svn` command in list form, suitable for passing to
         :class:`subprocess.Popen`.
     """
-    from . import __version__ as this_version
-    if desimodel_version is None:
-        export_version = 'trunk'
-    elif (desimodel_version == 'trunk') or ('branches/' in desimodel_version):
-        export_version = desimodel_version
+    svn_version = get_svn_version(desimodel_version)
+    if svn_checkout:
+        svn_subcommand = 'checkout'
     else:
-        export_version = 'tags/' + desimodel_version
-    return ["svn", "export",
-            ("https://desi.lbl.gov/svn/code/desimodel/" +
-             "{0}/data").format(export_version)]
+        svn_subcommand = 'export'
+
+    return ["svn", svn_subcommand,
+            f"https://desi.lbl.gov/svn/code/desimodel/{svn_version}/data"]
```

Comment:
I know it's been set up this way for a while, but I would recommend against hard-coding the URL into this function.

Reply:
What do you recommend instead?  A separate function dedicated to generating just that URL, to by used by this function that generates the svn command line?  Or is there some other way that we should be deriving the URL?

Reply:
I think an optional parameter to the function would be sufficient:
```python
def svn_export(desimodel_version=None, svn_checkout=False, svn_url='https://desi.lbl.gov/svn/code/desimodel'):
```